source/administration/incident-response-application.rst

```diff
@@ -0,0 +1,466 @@
+    "match", "The text string to match on", "string", "If ``match_regex`` not set, yes"
+    "match_regex", "The regex to match on", "string", "If ``match`` not set, yes"
+
+// TODO Verify JSON schema and add an example here.
+
+Step
+^^^^^^^^^^^^^^^
+
+A step is the representation of a state the app workflow can be in. For now it contains a set of actions to be performed when it is started and when it is finished.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "name", "The name of the step", "string", "Yes. This field must be unique."
+    "start_actions", "Actions to perform when the step is reached.", "[]Action", "If ``finished_actions`` not set, yes"
+    "finished_actions", "Actions to perform when the step is finished.", "[]Action", "If ``start_actions`` not set, yes"
+
+Action
+^^^^^^^^^^^^^^^
+
+Actions are performed when steps are started and finished.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "name", "The name of the action", "string", "Yes. This field must be unique."
+    "type", "The type of action to perform", "string", "Yes"
+    "*additional fields", "Additional fields based on the type of action. See below for more details.", "string", "Yes"
+
+Create Channel (type: ``create_channel``)
+*******************************************
+
+The create channel action creates a channel with the given parameters. The users listed will be automatically invited.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "channel", "A channel object supporting fields from model.Channel", "model.Channel", "Yes"
+
+// TODO Verify JSON schema and add an example here, with a clear recommendation for using instance template variables to avoid non-unique channel names. Example should include the supported fields of the channel model, e.g. type, team, ...
+
+Add Users to Channel (type: ``add_users_channel``)
+**************************************************************
+
+The add users to channel action adds the specified users to the channel.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "channel_name", "The channel to create the post in. Can be name or ID.", "model.Channel", "Yes"
+    "users", "Users to add to the channel after creation. Can be usernames, user IDs or AD/LDAP group names.", "[]string, "Yes"
+
+// TODO Verify JSON schema and add an example here.
+
+Create Post (type: ``post``)
+*******************************
+
+Creates a post in the specified channel.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "channel_name", "The channel to create the post in. Can be name or ID.", "string", "Yes"
+    "message", "The contents of the message.", "string", "Yes"
+    "fields", "A list of fields to show in the message. ", "[]Field", "No"
+    "transition_button", "A label and a step to transition to when pressed.", "[]Button", "No"
```

Comment:
We added more fields here. See https://github.com/mattermost/mattermost-plugin-workflow/blob/master/server/action/post.go#L17